src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -155,3 +156,17 @@ export const sumMainnetTokensUsdBalancesPerNetwork = ({
  */
 export const filterEnabledTokens = ([$tokens]: [$tokens: Token[]]): Token[] =>
 	$tokens.filter((token) => ('enabled' in token ? token.enabled : true));
+
+/** Pins enabled tokens at the top of the list, preserving the order of the parts.
+ *
+ * @param $tokens - The list of tokens.
+ * @returns The list of tokens with enabled tokens at the top.
+ * */
+export const pinEnabledTokensAtTop = <T extends Token>(
+	$tokens: TokenToggleable<T>[]
+): TokenToggleable<T>[] =>
+	$tokens
+		.reduce<
```

Comment:
Why not just a `.sort`?

Reply:
too used to use `reduce` eheh! simplified with `sort`